articles/100_ament.md

```diff
@@ -0,0 +1,176 @@
+---
+layout: default
+title: The meta build system "ament"
+permalink: articles/ament.html
+abstract:
+  This article describes the meta build system "ament".
+published: true
+author: '[Dirk Thomas](https://github.com/dirk-thomas)'
+---
+
+* This will become a table of contents (this text will be scraped).
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+
+## ament
+
+`ament` is a meta build system to build applications which are split into separate packages.
+
+It relies on meta information about the packages to determine their build dependencies.
+These meta information is specified in a file called `package.xml` which is defined in [REP 140](http://www.ros.org/reps/rep-0140.html).
+
+ament consists of several separate parts:
+
+* [ament_package](https://github.com/ament/ament_package) is a Python package providing an API to parse the files containing the meta information.
+* [ament_cmake](https://github.com/ament/ament_cmake) contains a set of packages which provide CMake based functionality.
+  The features are split across several packages in order to ensure that they are cleanly separated from each other.
+  The main feature of the CMake packages is to automatically generate a CMake config file to pass information to downstream packages.
+* [ament_lint](https://github.com/ament/ament_lint) contains a set of packages which perform linting tasks.
+  For as set of linters a ROS 2 specific configuration is provided which can be used via the command line as well as from within CMake (commonly as a CTest).
+* [ament_tools](https://github.com/ament/ament_tools) is a Python package which provide command line tools to build, test and install packages.
+  It is similar to [catkin_tools](https://github.com/catkin/catkin_tools) and builds each package in a workspace in topological order.
+  Each different build system is integrated through extension points.
+  While it currently does not build packages in parallel that will be added in the future to speed up the build process.
+* [ament_index](https://github.com/ament/ament_index) contains a set of packages which provide API to access the [ament resource index](https://github.com/ament/ament_cmake/blob/master/ament_cmake_core/doc/resource_index.md).
+  Currently it only contains a Python implementation but other languages like C++ should follow.
+
+
+### Integrate arbitrary build systems
+
+Each package can utilize a different build system to perform the steps of configuring, building and installing.
+Currently supported are CMake and Python but support for others (e.g. autotools, plain Makefiles) will likely be added in the future.
```

Comment:
I think this kind of differentiation is beyond the scope of this article. It should be part of the ament_tools documentation at some point.